ballerina/soap.bal

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2023, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/mime;
+
+# Soap version.
+public enum SoapVersion {
+    SOAP11,
+    SOAP12
+}
+
+# Soap client configurations.
+#
+# + soapVersion - Soap version
+public type ClientConfiguration record {|
+  *http:ClientConfiguration;
+  SoapVersion soapVersion = SOAP11;
+|};
+
+# Object for the basic SOAP client endpoint.
+public client class Client {
```

Comment:
```suggestion
public isolated client class Client {
```

Reply:
Fixed in [89df3b8](https://github.com/ballerina-platform/module-ballerina-soap/pull/82/commits/89df3b8ca3508f62ee3e4bcd88027135fa5babcb)